doc/source/dev/development_workflow.rst

```diff
@@ -223,9 +223,87 @@ these fragments in each commit message of a PR:
-  uploads.
-  `See the configuration file for these checks. <https://github.com/numpy/numpy/blob/main/.cirrus.star>`__
+  `CirrusCI <https://cirrus-ci.org/>`__ mostly triggers Linux aarch64 and MacOS
+  Arm64 wheels uploads. This is particularly expensive (see `gh-24280
+  <https://github.com/numpy/numpy/issues/24280>`__) and should be minimized.
+  `See the configuration file for these checks.
+  <https://github.com/numpy/numpy/blob/main/.cirrus.star>`__
+
+Additional CI Tags
+~~~~~~~~~~~~~~~~~~
+
+There are a range of additional tags that provide more fine-grained control over
+CI jobs. These tags allow are meant to skip or include specific jobs that are
+not needed for your changes. Here are the new tags you can use:
+
+* ``[skip/only circle]``: skip / run only the CircleCI jobs (documentation and preview mostly)
+* ``[skip/only linux]``: skip / run only Linux jobs, this is typically not a good idea unless another specific CI run is requested
+* ``[skip/only nixblas]``: skip / run only BLAS NumPy jobs
+* ``[skip/only sanitizer]``: skip / run only compiler sanitizer jobs (Linux)
+* ``[skip/only musl]``: skip / run only musllinux_x86_64 jobs
+* ``[skip/only qemu]``: skip / run only Linux QEMU jobs
+* ``[skip/only simd]``: skip / run only SIMD optimization jobs
+* ``[skip/only macos]``: skip / run only MacOS jobs
+* ``[skip/only mypy]``: skip / run only ``mypy`` type checking jobs
+* ``[skip/only wingha]``: skip / run only Windows GHA jobs
+* ``[skip/only wasm]``: skip / run only Emscripten/Pyodide jobs
+* ``[skip/only codeql]``: skip / run only CodeQL analysis jobs
+
+These tags do not need to be in the first line of the commit message and can
+appear anywhere in the body of the commit itself. Additionally, they can be used
+in any order, i.e. ``[skip mypy]`` is the same as ``[mypy skip]``.
+
+Example Usage
+-------------
+
+An example commit message to run only Windows F2PY tests and skip other jobs:
+
+.. code-block:: text
+
+   CI: Configure F2PY GHA Windows [only winf2py]
+
+   [skip circle] [skip linux] [skip nixblas] [skip sanitizer] [skip musl]
+   [skip qemu] [skip simd] [skip macos] [skip mypy] [skip wingha] [wasm skip]
+   [codeql skip]
+
+   [skip azp]
+   [skip cirrus]
+
+Note: The ``[skip azp]`` tag can be anywhere in the commit message,
+but ``[skip cirrus]`` must be in the first or last line.
+
+Helper (compound) tags
+----------------------
+
+Since the tags can be a bit of a mouthful, there are some short-hand helpers,
+which expand to the above.
+
+* ``[skip gha]``: Skips everything on GithubActions except for ``[only ]`` directives
+* ``[skip except]``: Skips everything except for ``[only ]`` directives
+* ``[docs only]``: Skips everything but the documentation job
+* ``[skip exotic]``: ``[skip quemu] [skip simd] [skip macos] [skip wingha] [skip wasm] [skip musl] [skip circle] [skip azp] [ckip cirrus]``
+* ``[skip tools]``: Expands to ``[skip codeql] [skip mypy] [skip sanitizer]``
+
+Being CI Sensitive
```

Comment:
My suggestion: maybe this can be an admonition? My only concern is that this maybe too complicated for a regular contributor and we often point them to this section to see the skip commands. Being in an admonition may change the flow of the text so they realize this is more of an advanced workflow.